R/rc.get.xmcs.data.R

```diff
@@ -136,7 +136,7 @@ rc.get.xcms.data <- function(xcmsObj = NULL,
       nfiles <- nrow(xcmsObj@phenoData)
     }
     if (newXCMS) {
-      nfiles <- length(xcmsObj@phenoData[[1]])
+      nfiles <- length(xcmsObj@phenoData@data[[1]])
```

Comment:
Also here, it's better to use `inherits()` to check which object class you have. And possible fix here:

```r
if (inherits(xcmsObj, "XCMSnExp")) {
    nfiles <- nrow(MSnbase::pData(xcmsObj))
}
if (inherits(xcmsObj, "XcmsExperiment")) {
    nfiles <- MsExperiment::length(xcmsObj)
}
```